code/datums/gameticker.dm

```diff
@@ -368,6 +370,40 @@ var/global/current_state = GAME_STATE_WORLD_INIT
 				if(player.mind.assigned_role != "MODE")
 					player.Equip_Rank(player.mind.assigned_role)
 
+	///Generates a list of all picky eaters then assigns them 5 favorite foods
+	proc/handle_picky_eaters()
```

Comment:
This is mostly duplicated on `/mob/new_player`, and having this on the `gameticker` anyways seems weird.
Would prefer if you moved this proc to the trait datum itself (and adjusted vars accordingly), and just put all this code in the trait's `New()` proc. I think that traits aren't actually recreated on cloning, just transferred? Since the allergy trait sets itself up in New() and that doesn't change between clonings.

Reply:
I moved the proc to the trait's "onadd" proc. However i ran into an issue. The onadd is ran before the mind is placed inside the mob, so it makes it difficult to boutput the food list to the player, or to store the foodlist in their mind. I left some code in new_player and gameticker since i would ideally like the fav food list to only appear once on gamestart and now on every New() human. If this is inefficient please let me know!